### ISSUE TITLE
Fix `TestCustomToolWithGitCreds`

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-cd#2090]
Fails locally.

```
=== RUN   TestCustomToolWithGitCreds
time="2019-08-02T13:48:41-07:00" level=info msg="kubectl delete ns -l e2e.argoproj.io=true --field-selector status.phase=Active --wait=false" dir= execID=nZlKy
time="2019-08-02T13:48:41-07:00" level=debug msg="namespace \"argocd-e2e-test-cluster-role-binding-mblqn\" deleted\n" duration=86.884187ms execID=nZlKy
time="2019-08-02T13:48:41-07:00" level=info msg="Starting configmap/secret informers"
time="2019-08-02T13:48:41-07:00" level=info msg="secrets informer cancelled"
time="2019-08-02T13:48:41-07:00" level=info msg="configmap informer cancelled"
time="2019-08-02T13:48:41-07:00" level=info msg="Configmap/secret informer synced"
time="2019-08-02T13:48:42-07:00" level=info msg="mkdir -p /tmp/argo-e2e" dir= execID=evTZR
time="2019-08-02T13:48:42-07:00" level=debug duration=4.36644ms execID=evTZR
time="2019-08-02T13:48:42-07:00" level=info msg="mkdir -p /tmp/argo-e2e/app/config/tls" dir= execID=lr3YG
time="2019-08-02T13:48:42-07:00" level=debug duration=4.301133ms execID=lr3YG
time="2019-08-02T13:48:42-07:00" level=info msg="mkdir -p /tmp/argo-e2e/app/config/ssh" dir= execID=WoeZK
time="2019-08-02T13:48:42-07:00" level=debug duration=5.664229ms execID=WoeZK
time="2019-08-02T13:48:42-07:00" level=info msg="cp -Rf testdata /tmp/argo-e2e/testdata.git" dir= execID=D64fa
time="2019-08-02T13:48:42-07:00" level=debug duration=52.96723ms execID=D64fa
time="2019-08-02T13:48:42-07:00" level=info msg="chmod 777 ." dir=/tmp/argo-e2e/testdata.git execID=MtA8z
time="2019-08-02T13:48:42-07:00" level=debug duration=4.099405ms execID=MtA8z
time="2019-08-02T13:48:42-07:00" level=info msg="git init" dir=/tmp/argo-e2e/testdata.git execID=zpECN
time="2019-08-02T13:48:42-07:00" level=debug msg="Initialized empty Git repository in /private/tmp/argo-e2e/testdata.git/.git/\n" duration=19.210781ms execID=zpECN
time="2019-08-02T13:48:42-07:00" level=info msg="git add ." dir=/tmp/argo-e2e/testdata.git execID=HMj7o
time="2019-08-02T13:48:42-07:00" level=debug duration=58.897646ms execID=HMj7o
time="2019-08-02T13:48:42-07:00" level=info msg="git commit -q -m initial commit" dir=/tmp/argo-e2e/testdata.git execID=9Ncc4
time="2019-08-02T13:48:42-07:00" level=debug duration=66.722425ms execID=9Ncc4
time="2019-08-02T13:48:42-07:00" level=info msg="kubectl create ns argocd-e2e-test-custom-tool-with-git-creds-mkgyo" dir= execID=AnZMr
time="2019-08-02T13:48:43-07:00" level=debug msg="namespace/argocd-e2e-test-custom-tool-with-git-creds-mkgyo created\n" duration=103.835454ms execID=AnZMr
time="2019-08-02T13:48:43-07:00" level=info msg="kubectl label ns argocd-e2e-test-custom-tool-with-git-creds-mkgyo e2e.argoproj.io=true" dir= execID=m6J7K
time="2019-08-02T13:48:43-07:00" level=debug msg="namespace/argocd-e2e-test-custom-tool-with-git-creds-mkgyo labeled\n" duration=134.104089ms execID=m6J7K
time="2019-08-02T13:48:43-07:00" level=info msg="clean state" duration=1.93467269s id=test-custom-tool-with-git-creds-mkgyo name=test-custom-tool-with-git-creds
time="2019-08-02T13:48:43-07:00" level=info msg="../../dist/argocd cert add-tls localhost --from /Users/acollins8/go/src/github.com/argoproj/argo-cd/test/fixture/certs/argocd-test-ca.crt --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=JKtvE
time="2019-08-02T13:48:43-07:00" level=debug msg="Reading TLS certificate data in PEM format from '/Users/acollins8/go/src/github.com/argoproj/argo-cd/test/fixture/certs/argocd-test-ca.crt'\nCreated entry with 1 PEM certificates for repository server localhost\n" duration=176.674117ms execID=JKtvE
time="2019-08-02T13:48:43-07:00" level=info msg="../../dist/argocd cert add-tls 127.0.0.1 --from /Users/acollins8/go/src/github.com/argoproj/argo-cd/test/fixture/certs/argocd-test-ca.crt --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=dReR2
time="2019-08-02T13:48:43-07:00" level=debug msg="Reading TLS certificate data in PEM format from '/Users/acollins8/go/src/github.com/argoproj/argo-cd/test/fixture/certs/argocd-test-ca.crt'\nCreated entry with 1 PEM certificates for repository server 127.0.0.1\n" duration=164.736493ms execID=dReR2
time="2019-08-02T13:48:43-07:00" level=info msg="../../dist/argocd repo add https://localhost:9443/argo-e2e/testdata.git --username admin --password password --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure" dir= execID=OD5DJ
time="2019-08-02T13:48:43-07:00" level=debug duration=60.748752ms execID=OD5DJ
time="2019-08-02T13:48:43-07:00" level=error msg="`../../dist/argocd repo add https://localhost:9443/argo-e2e/testdata.git --username admin --password password --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure` failed exit status 1: time=\"2019-08-02T13:48:43-07:00\" level=fatal msg=\"rpc error: code = Unknown desc = Get https://localhost:9443/argo-e2e/testdata.git/info/refs?service=git-upload-pack: x509: certificate signed by unknown authority\"" execID=OD5DJ
time="2019-08-02T13:48:43-07:00" level=fatal msg="`../../dist/argocd repo add https://localhost:9443/argo-e2e/testdata.git --username admin --password password --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3Nzg3MDQsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3ODcwNCwic3ViIjoiYWRtaW4ifQ.ddiLrG_s7hSvVTrSTgmFNIkoLduyvK3_j9HGIAubWfg --insecure` failed exit status 1: time=\"2019-08-02T13:48:43-07:00\" level=fatal msg=\"rpc error: code = Unknown desc = Get https://localhost:9443/argo-e2e/testdata.git/info/refs?service=git-upload-pack: x509: certificate signed by unknown authority\""
FAIL	github.com/argoproj/argo-cd/test/e2e	219.725s
```

**Comment by alexec** [comment]
@jannfis @alexmt this test only fails locally with the error:

```
time="2019-08-02T13:52:28-07:00" level=fatal msg="`../../dist/argocd repo add https://localhost:9443/argo-e2e/testdata.git --username admin --password password --plaintext --server localhost:8080 --auth-token eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE1NjQ3NzkxNDUsImlzcyI6ImFyZ29jZCIsIm5iZiI6MTU2NDc3OTE0NSwic3ViIjoiYWRtaW4ifQ.48w-VJjPdLBLDWUuMRm0mjFYMtPwRjSMZT-o_WLsOwI --insecure` failed exit status 1: time=\"2019-08-02T13:52:28-07:00\" level=fatal msg=\"rpc error: code = Unknown desc = Get https://localhost:9443/argo-e2e/testdata.git/info/refs?service=git-upload-pack: x509: certificate signed by unknown authority\""
``

Thoughts?

**Comment by alexec** [comment]
`TestCanAddAppFromPrivateRepoWithCredCfg` also fails, but only when run on its own. The mystery continues...

**Comment by jannfis** [comment]
Fixed by PR #2097 - didn't notice on my dev box because `/app/config` was writeable by my user ID. CircleCI execution is not affected because the end-2-end test server is started in a different way, not using `Makefile`.
